### PR TITLE
TDL-16349: Add request timeout

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-16349: Add request timeout
- Use default timeout of 300 seconds if the user does not enter the preferred timeout value

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,7 +48,8 @@ This tap:
     ```json
```
Comment (umangagrawal-crest) [inline README.md:48]: Add the description for the request_timeout in point 3

**Review by umangagrawal-crest — CHANGES_REQUESTED** [review]
Do we need to specify the dependency of backoff in the setup.py file ??

File: README.md (hunk 1 above)
Comment (harshpatel4crest) [reply README.md:48]: Added

**Comment by harshpatel4crest** [comment]
> Do we need to specify the dependency of backoff in the setup.py file ??

When we import `singer-python`, backoff will be imported. [Reference](https://github.com/singer-io/singer-python/blob/0c066de21111d8572425083b4a8792d193c80af1/setup.py#L17)

File: tap_github/__init__.py [hunk 2]
```diff
@@ -1010,6 +1017,20 @@ def get_stream_from_catalog(stream_id, catalog):
...
     return None
 
+# return the 'timeout'
+def get_request_timeout():
+    args = singer.utils.parse_args([])
+    # get the value of request timeout from config
+    config_request_timeout = args.config.get('request_timeout')
+
+    if config_request_timeout and float(config_request_timeout):
+        request_timeout = float(config_request_timeout)
```
Comment (savan-chovatiya) [inline tap_github/__init__.py:1027]: Add detailed comment above `if` condition.

File: tap_github/__init__.py [hunk 3]
```diff
@@ -1010,6 +1017,20 @@ def get_stream_from_catalog(stream_id, catalog):
...
+        # return the timeout from config
+        return request_timeout
```
Comment (savan-chovatiya) [inline tap_github/__init__.py:1029]:
```suggestion
        return float(config_request_timeout)
```

File: tap_github/__init__.py (hunk 2 above)
Comment (harshpatel4crest) [reply tap_github/__init__.py]: Added comment.

File: tap_github/__init__.py (hunk 3 above)
Comment (harshpatel4crest) [reply tap_github/__init__.py]: Updated.

**Review by savan-chovatiya — APPROVED** [review]

**Review by umangagrawal-crest — APPROVED** [review]

**Review by namrata270998 — APPROVED** [review]

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**